test/missing.jl

```diff
@@ -193,11 +193,15 @@ end
     # All rounding functions return missing when evaluating missing as first argument
     for f in rounding_functions
         @test ismissing(f(missing))
-        @test ismissing(f(missing, 1))
-        @test ismissing(f(missing, 1, 1))
+        @test ismissing(f(missing, digits=1))
+        @test ismissing(f(missing, digits=1, base=1))
         @test ismissing(f(Union{Int, Missing}, missing))
         @test f(Union{Int, Missing}, 1.0) === 1
         @test_throws MissingException f(Int, missing)
+        @test ismissing.(f.([1.0, missing])) == [false, true]
```

Comment:
I don't think these tests are useful. `f` is called either on `missing` which is covered above, or on `1.0` which is covered elsewhere. Please remove them to keep the tests simple and similar to what is done for other operations.

Reply:
Do you just mean line 201, or lines 201-204? The reason I added these tests was to prevent the Number and Missing versions of these functions from getting out of sync again, so I call them elementwise with the various argument combinations. If the Number definitions change again, this *might* catch it. This is how I interpreted the request to "exercise this codepath", but I'm happy to drop these tests if they're unwelcome. I guess we're unlikely to see the same problem again!

Reply:
OK, let's keep these then. I'm afraid other functions in this file would need the same treatment, but I guess covering part of them is better than nothing.

Reply:
For the record, not testing any given Missing functionality on mixed arrays is rather like not testing a car on roads. It looks like there are a lot of other elementwise tests in the file.

Reply:
Writing good tests is very hard. Ideally we should only test the elementary blocks to avoid combinatorial explosion of tests. I can be a good idea to test a few common patterns too, but in the present case the `f.(...)` really just applies `f` to each element, so that's nothing more than a compact way of applying `f` to each entry and checking the result.

Reply:
Yes, but if you only test these functions on scalars, you can get the problem that led to this PR: The signature of `round(x::Float...)` changed, but `round(x::Missing...)` was forgotten (and so were its tests, naturally). Because tests on mixed `Union{Float..., Missing}` arrays did not exist, this problem did not show up as a test failure. Since the whole point of `Missing` is to support these mixed arrays, I think that was a gap in the tests, which these new tests fill in. Note that there are many other tests for mixed arrays in the `test/missing.jl`.

Reply:
Again, arrays are *not* involved in this code. What matters is that `round` is called with the same arguments on both `missing` and non-`missing` values. We could do that on separate lines if we wanted, the result would be the same.

Reply:
Yes, I understand that, but isn't it nice to have it in one line? 😃 More seriously, it would seem weird (to me) to check the `Float` function twice, here and in the float tests (although I realize this is what's happening behind the scenes). On the other hand, it seems worthwhile to test that the signatures are compatible in order to avoid the kind of problem that triggered this PR. I thought this would be a good way to do that. Do you know of a nicer way to do this? 

Perhaps I should add a comment explaining why these tests are present...

Reply:
I have removed duplicated tests of Missing functionality and added some comments.

Reply:
I'm not opposed to doing this that way, I just want to make it clear what that line does.

Reply:
Okay, I see. Please check the latest commit, which I think cleans things up quite a bit.